### ISSUE TITLE
Jacobian

### ISSUE THREAD

**Issue by nourgaliev** [JamesYang007/FastAD#101]
Can we get any example when we have an array of functions, rather than a single one. This is quite a common case, when we need to compute Jacobian. I think I have some idea how to do that, but I am not sure this would be efficient.

**Comment by sofusmo** [comment]
Yeah - this seems to be a weak point of this library.   Doing a Jacobian for complex calculations is one place where an effort like Adept I would expect to clearly be faster, as the adjoint evaluation is a separate final step. 

It doesn't seem to be easy to change.  @JamesYang007 do you have suggestions for how to improve this?
